pinot-common/src/test/java/com/linkedin/pinot/common/utils/TarGzCompressionUtilsTest.java

```diff
@@ -122,4 +128,61 @@ public void testEmptyDirectory()
     Assert.assertEquals(segmentFiles.length, 0);
 
   }
+
+  @Test
+  public void testBadFilePath() throws Exception {
+    File metaFile = new File(segmentDir, "metadata.properties");
+    metaFile.createNewFile();
+
+    // We guarantee metaFile isn't a directory.
+    Assert.assertFalse(metaFile.isDirectory());
+
+    File tarGzPath = new File(tarDir, SEGMENT_NAME + ".tar.gz");
+    mockIncorrectData(metaFile, tarGzPath);
+
+    try {
+      TarGzCompressionUtils.unTar(tarGzPath, untarDir);
+      Assert.fail("Did not get exception!!");
+    } catch (IOException e) {
+      Assert.assertTrue(e.getMessage().contains(TarGzCompressionUtils.EXTRACT_FILE_OUTSIDE_OF_TARGET_DIR));
+    }
+  }
+
+  private void mockIncorrectData(File nonDirFile, File tarGzPath) throws IOException {
```

Comment:
why are we naming this mockIncorrectData? can we call it something like createInvalidTarFile